### DISCUSSION TITLE
UI improvements

### DISCUSSION THREAD

**Discussion by erengy** [erengy/taiga#1066] [Ideas]
This discussion is intended to provide an overview of ideas to improve Taiga's user interface, which might help with our evaluation of UI frameworks (#1050). Feel free to post other ideas or inconveniences you've had with the UI.

## General

- Better high-DPI support
  - #253
  - #1017
- Dark mode
  - #304
  - #320
  - #679
  - #685
  - #762
  - #807
  - #926
  - #980
  - #1002
- #344
- #665
- #838

## Main window

- #237
- #364

## Now Playing

- #715

## Anime List

- #87
- #102
- #150
- #235
- #247
- #533
- https://github.com/erengy/taiga/issues/576#issuecomment-539145046 (Custom progress bar colors)
- #847
- #1093
- #1101
- #1137

## Anime Information

- #132
- #362
- #636 
- #853
- #948
- #954

## History

- #1118

## Seasons

- #790
- #1108

## Torrents

- #607

**Comment by Paraxnoir** [comment]
Hello @erengy!

I posted on your [Discord channel](https://discord.com/channels/423475967051169813/423475967051169815/1061353575009226952) but was directed here for improvement proposals.
![image](https://user-images.githubusercontent.com/39978965/211175240-d4835646-1915-4fb0-ab9a-ad3db6df94e0.png)

Would it be possible to be manually able to color code anime directly and exclusively on Taiga?

Let us take as an example my current watching list, which tends to be a bit chaotic. I quickly added in two color codes, which would help my poor mind not wander too far off, and focus on what I really want to prioritize watching/finishing at the time.

If at all doable, it would be highly appreciated, and I think an optional feature that many would end up using!

Just want to add I'm a long time user and love Taiga. Thank you again for all the work you've done! Take care ++

**Reply by erengy** [reply]
Hi there @Paraxnoir, thanks for your suggestion.

From what I understand, your request is similar to #247 in the sense that they are both about priorities. It seems color coding was mentioned at #820 too, so it might be a useful addition indeed.

My initial reaction is that painting entire rows in different colors would make the list look like a spreadsheet, and having too much color would eventually detract from the UX. Not to mention that it's already being used for a different purpose, which is to indicate currently playing anime with a light green background.

Perhaps something like macOS's tags would work as an alternative solution that makes use of colors but in moderation. It's also general enough so that it can be used for other purposes. But there are still other concerns, such as:

- Not sure if the colors should be pre-made or customizable.
- Colored icons are already being used to indicate airing status.
- We need to consider how this would all look in dark mode too.
- We need to make it obvious that it's client-side only.

Color-coded rows might not be it, but I do think that we should find a way to add a similar feature.

**Reply by Paraxnoir** [reply]
Thank you for the rapid and detailed answer! I believe colored tags akin to macOS would work great, and certainly look more sleek than the highlighter I added to my screenshot.

Regarding the colors themselves, customizable is always good, but if a few pre-made colors work out better and allow easier implementation of the requested dark theme, I believe this is more than sufficient for the purposes of having a visual for priorities/tagging/categories.

Perhaps the tags idea could work out as a colored dot at the end of an anime title. This ensures a clear separation from the airing status, while not necessarily requiring a new column (though maybe new column is a better idea).
Perhaps the tag idea can grow to be more than a mere color code, with its own related section where users can build local personalised lists using tags.
Perhaps the tags can be more than just colors..? Words also?

These are the ideas off the top of my head. Thank you for your consideration!

**Comment by ekk88d** [comment]
This one would be great, and hopefully not too difficult to implement:

- Anime Information window: Make the Producers, Season, Genre etc., linked to their corresponding values on MAL. If the user previously selected a different metadata provider, the links can go to those sites' sections, if available.
This would be really useful in cross-referencing and finding similar anime. Maybe this also spills into one of your other projects.

- Anime information window: Title field links directly to metadata provider's page for that anime. I know that users can use right-click > search to find anime listings from the list view, but it's quicker to have that direct link onhand.

While that'd be ideal, I've been working around it by copy/pasting each anime's direct link in their Anime Information >  Notes: section.
As an altenative, right below the Notes: field could be one for the user-entered MAL/AniList/Kitsu page link. That'd eliminate needing to directly map the links.

Thank you for your ongoing development!

**Reply by erengy** [reply]
> Anime Information window: Make the Producers, Season, Genre etc., linked to their corresponding values on MAL.

Having more links around should be possible in v2 with Qt's widgets. There's going to be other improvements too such as anime details not being limited to a single line or making the window resizable.

For future reference, genres should be linked as following:

- AniList: `https://anilist.co/search/anime?genres={title}`
- Kitsu: `https://kitsu.io/anime?categories={slug}`
- MAL: `https://myanimelist.net/anime/genre/{id}/`

Note to self: This may require us to store more data (e.g. IDs) about genres, producers, etc.

> If the user previously selected a different metadata provider, the links can go to those sites' sections, if available.

Switching metadata providers basically means resetting Taiga's database, so I'm afraid that won't be possible. In any case, this goes much beyond the scope of UI.

> Anime information window: Title field links directly to metadata provider's page for that anime. I know that users can use right-click > search to find anime listings from the list view, but it's quicker to have that direct link onhand.

In case you missed it, you can already click the poster image for that. Perhaps we can add some kind of hover state to the image to make this more obvious. You can also set up *Settings → Application → Anime List → Actions* to go directly from Anime List to the anime page.
